react-app/src/components/ProposalScreen/ProposalScreen.tsx

```diff
@@ -41,75 +37,75 @@ const defaultFilterItems: IFilterTabItem<FilterKey>[] = [
+
 const ProposalScreen: React.FC = () => {
   const wallet = useWallet();
   const { translate } = useLocale();
-  const [searchParams, setSearchParams] = useSearchParams();
+  const [selectedTab, setSelectedTab] = useState<FilterKey>(
+    defaultTabItem.value
+  );
+  const [after, setAfter] = useState(0);
 
-  const [offset, filter, searchTerm] = useMemo(() => {
-    const pageStr = searchParams.get("page") ?? "1";
-    const filter = searchParams.get("filter") ?? "voting";
-    const search = searchParams.get("search");
+  const [searchParams, setSearchParams] = useSearchParams({
+    first: PROPOSAL_LIST_PAGE_SIZE.toString(),
+    after: after.toString(),
+    tab: defaultTabItem.value,
+  });
 
-    const page = Math.max(parseInt(pageStr, 10), 1) - 1;
+  const tabItems: ProposalTabItem[] = useMemo(() => {
+    if (wallet.status === ConnectionStatus.Connected) {
+      const followingTab = {
+        value: "following" as const,
+        label: "ProposalScreen.filters.following" as const,
+      };
+      return [...defaultTabItems, followingTab];
+    }
 
-    return [page * PROPOSAL_LIST_PAGE_SIZE, filter as FilterKey, search];
-  }, [searchParams]);
+    return defaultTabItems;
+  }, [wallet]);
 
   const { requestState, fetch } = useProposalsQuery(
-    offset,
+    after,
     PROPOSAL_LIST_PAGE_SIZE
   );
 
-  const filterItems: IFilterTabItem<FilterKey>[] = useMemo(() => {
-    if (wallet.status === ConnectionStatus.Connected) {
-      const followingTab: IFilterTabItem<FilterKey> = {
-        label: "ProposalScreen.filters.following",
-        value: "following",
-      };
-      return [...defaultFilterItems, followingTab];
-    }
+  const setPage = useCallback((after: number) => {
+    setAfter(after);
+  }, []);
 
-    return defaultFilterItems;
-  }, [wallet]);
+  const handleSelectTab = useCallback((tab: FilterKey) => {
+    setSelectedTab(tab);
+  }, []);
 
-  const setPage = useCallback(
-    (offset: number) => {
-      const page = Math.floor(offset / PROPOSAL_LIST_PAGE_SIZE) + 1;
-      const params = searchParams;
-      params.set("page", page.toString());
-      setSearchParams(params);
-    },
-    [searchParams, setSearchParams]
-  );
+  useEffect(() => {
+    setSearchParams({
+      first: PROPOSAL_LIST_PAGE_SIZE.toString(),
+      after: after.toString(),
+      tab: selectedTab,
+    });
+    fetch({
+      first: PROPOSAL_LIST_PAGE_SIZE,
+      after,
+      tab: selectedTab,
+    });
+  }, [after, fetch, selectedTab, setSearchParams]);
 
-  const setFilter = useCallback(
-    (filter: FilterKey) => {
-      const params = searchParams;
-      params.set("filter", filter);
-      params.set("page", "1");
-      setSearchParams(params);
-    },
-    [setSearchParams, searchParams]
-  );
+  useEffect(() => {
+    const tab = tabItems.find((item) => item.value === searchParams.get("tab"));
+    if (tab) {
+      setSelectedTab(tab.value);
+    }
+  }, [fetch, handleSelectTab, searchParams, tabItems]);
```

Comment:
would this cause the query to fire multiple times?

Reply:
yes, I should empty the deps array so this will only fire on first load of the page